src/read/coff/import.rs

```diff
@@ -0,0 +1,208 @@
+    fn from_u2(n: u16) -> Result<Self> {
+        match n {
+            pe::IMPORT_OBJECT_CODE => Ok(Self::Code),
+            pe::IMPORT_OBJECT_DATA => Ok(Self::Data),
+            pe::IMPORT_OBJECT_CONST => Ok(Self::Const),
+            0b11 => Err(Error("Invalid import type")),
+            _ => unreachable!("ImportType must be a two bit number"),
+        }
+    }
+}
+
+/// Determines how the symbol is imported.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub enum ImportNameType {
+    /// Import by an offset into the DLL's export table.
+    Ordinal,
+    /// Import using the same name as the symbol name.
+    Name,
+    /// Strip any leading `?`, `@` or `_` from the symbol name to get the import name.
+    NameNoPrefix,
+    /// The same as [`NameNoPrefix`] but also truncates at the next `@`.
+    NameUndecorate,
+    /// The import name is given after the DLL name.
+    /// This is undocumented and may not be supported by many tools.
+    NameExportAs,
+}
+impl ImportNameType {
+    /// `n` must be a three bit number.
+    fn from_u3(n: u16) -> Result<Self> {
+        match n {
+            pe::IMPORT_OBJECT_ORDINAL => Ok(ImportNameType::Ordinal),
+            pe::IMPORT_OBJECT_NAME => Ok(ImportNameType::Name),
+            pe::IMPORT_OBJECT_NAME_NO_PREFIX => Ok(ImportNameType::NameNoPrefix),
+            pe::IMPORT_OBJECT_NAME_UNDECORATE => Ok(ImportNameType::NameUndecorate),
+            pe::IMPORT_OBJECT_NAME_EXPORTAS => Ok(ImportNameType::NameExportAs),
+            5..=7 => Err(Error("Unknown name type")),
+            _ => unreachable!("ImportNameType must be a three bit number"),
+        }
+    }
+}
+
+impl pe::ImportObjectHeader {
+    /// Read the short import header.
+    ///
+    /// Also checks that the signature and version are valid.
+    /// Directly following this header will be the string data.
+    pub fn parse<'data, R: ReadRef<'data>>(data: R, offset: &mut u64) -> Result<&'data Self> {
+        let header = data
+            .read::<crate::pe::ImportObjectHeader>(offset)
+            .read_error("Invalid import header size")?;
+        if header.sig1.get(LE) != 0 || header.sig2.get(LE) != pe::IMPORT_OBJECT_HDR_SIG2 {
+            Err(Error("Invalid import header"))
+        } else if header.version.get(LE) != 0 {
+            Err(Error("Unknown import header version"))
+        } else {
+            Ok(header)
+        }
+    }
+
+    /// Get the machine type.
+    pub fn architecture(&self) -> Architecture {
+        match self.machine.get(LE) {
+            pe::IMAGE_FILE_MACHINE_ARMNT => Architecture::Arm,
+            pe::IMAGE_FILE_MACHINE_ARM64 => Architecture::Aarch64,
+            pe::IMAGE_FILE_MACHINE_I386 => Architecture::I386,
+            pe::IMAGE_FILE_MACHINE_AMD64 => Architecture::X86_64,
+            _ => Architecture::Unknown,
+        }
+    }
+
+    /// The type of import can be code, data or const.
+    pub fn import_type(&self) -> Result<ImportType> {
```

Comment:
I prefer if methods on raw types like `pe::ImportObjectHeader` only return raw values. So in this case it should return `u16` instead of `ImportType`. Similarly for `name_type`. Similarly we shouldn't return `Architecture` from a method for this type; we could change `architecture` to `machine`, but I don't think that provides enough value.